### ISSUE TITLE
关于deepspeed  模式进行ppo训练的问题-》显存占用超额

### ISSUE THREAD

**Issue by 2043078895** [modelscope/ms-swift#3210]
如题。
目前正在尝试deepspeed的zero2和3训练模型，核心关注的参数应该是以下几个：
NPROC_PER_NODE=4
CUDA_VISIBLE_DEVICES=0,1,2,3
--deepspeed zero2 \
--train_type full \
--load_args false \
--max_length 1000 \
--weight_decay 0.00001 \
--num_train_epochs 1 \
--torch_dtype bfloat16 \
--learning_rate 5e-5 \
--attn_impl flash_attn \
--truncation_strategy delete \
--gradient_checkpointing true \
--per_device_train_batch_size 1 \
--gradient_accumulation_steps 1 \
--gradient_checkpointing_kwargs "{\"use_reentrant\": false}" \
目前加载的奖励model为7B大小，policy model为1.5B大小，4张A100卡（40GB每张），一般而言训练这个模型绝对够用的对吧，可结果直接爆显存了，我推测应该是数据并行导致模型显存占用过大，那请问如何关闭ddp模式呢？
如果推测不对，麻烦看下是什么问题。

**Comment by 2043078895** [comment]
swift版本3.0.3，更高的版本报更多奇怪的错误，这个版本至少还能走到显存爆炸这

**Comment by 2043078895** [comment]
@Jintao-Huang 老哥，麻烦关注下这个问题

**Comment by hjh0119** [comment]
try smaller reward model or use peft(lora)

**Comment by 2043078895** [comment]
> try smaller reward model or use peft(lora)
如果是标准的ppo，显存没道理不够的哦，我用zero3也没跑起来。另外每个进程独立副本就算跑不起来，出现“Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cuda:3!”的问题看起来是swift的变量的管理上出的问题
